demos/TransferLearning/TransferLearningDemo.py

```diff
@@ -0,0 +1,481 @@
+                                      description='Model:')
+    ngraph_dropdown = widgets.Dropdown(options=ngraph_backends,
+                                       value=ngraph_backends[0],
+                                       description='nGraph:')
+
+    progress_bar = widgets.IntProgress()
+    progress_text = widgets.Label()
+    epoch_text = widgets.Label()
+    progress_box = widgets.HBox([progress_bar, progress_text, epoch_text])
+
+    batch_slider = widgets.IntSlider(min=1, max=100, value=batch_size, description='Batch Size:')
+    epoch_slider = widgets.IntSlider(min=1, max=16, value=epochs, description='Epochs:')
+    fine_tune_slider = widgets.IntSlider(min=1,
+                                         max=100,
+                                         value=fine_tune_at,
+                                         description='Fine Tune at Layer:')
+
+    train_button = widgets.Button(description='Train', disabled=True)
+    classify_button = widgets.Button(description='Classify', disabled=True)
+    fine_tune_button = widgets.Button(description='Fine Tune', disable=True)
+    shuffle_button = widgets.Button(description='Shuffle')
+
+    training_tab = VBox(children=[
+        model_dropdown, ngraph_dropdown, batch_slider, epoch_slider, train_button, classify_button,
+        shuffle_button
+    ])
+    fine_tuning_tab = VBox(children=[
+        batch_slider, epoch_slider, fine_tune_slider, fine_tune_button, classify_button,
+        shuffle_button
+    ])
+    tab = widgets.Tab(children=[training_tab, fine_tuning_tab])
+    tab.set_title(0, 'Training')
+    tab.set_title(1, 'Fine Tuning')
+    gui = VBox(children=[tab, progress_box, out_initial, out])
+
+    def __init__(self, verbose=0):
+
+        self.verbose = verbose
+        if verbose:
+            self.gui = VBox(
+                children=[self.tab, self.progress_box, self.out_initial, self.out, self.out_stats])
+
+        # Images
+        self.test_class_indices = []
+        self.test_dir = None
+        self.init_images()
+
+        # Model
+        #self.base_model = ResNet50(input_shape=self.IMAGE_SHAPE,include_top=False,weights='imagenet')
+        #self.base_model.trainable = False
+
+        #self.model = tf.keras.Sequential([
+        #  self.base_model,
+        #  keras.layers.GlobalAveragePooling2D(),
+        #  keras.layers.Dense(self.NUM_CLASSES, activation=self.DENSE_LAYER_ACTIVATION)
+        #])
+
+        self.sgd = optimizers.SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
+        #self.model.compile(optimizer = self.sgd, loss = self.OBJECTIVE_FUNCTION, metrics = self.LOSS_METRICS)
+
+        self.predicted_class_indices_init = []
+        self.wrong_guesses = []
+
+        display(self.init_gui())
+        self.train_button.disabled = False
+        self.fine_tune_button.disabled = False
+
+        self.init_model()
+
+    def init_images(self):
+        zip_file = tf.keras.utils.get_file(
```

Comment:
Just curious -- who owns this bucket?  Is it going to remain stable?

Reply:
The dataset is a subset of Kaggle's cats and dogs dataset; this zip is referenced by the tensorflow tutorial. Will look into maintaining our own dataset though.